common/serde-reflection/tests/serde.rs

```diff
@@ -2,44 +2,77 @@
 // SPDX-License-Identifier: Apache-2.0
```

Comment:
Can you add a test which tests that the tracing can still handle types which share a common name (or write a test which fails because we currently can't handle this)

Reply:
It's going to try to merge the types and (likely) fail.
In the long run, this is going to be an issue for polymorphic types, actually. We may have to index the registry using a more precise key (`T::type_name()` ?)

Reply:
Yeah using [`type_name()`](https://doc.rust-lang.org/stable/std/any/fn.type_name.html) would be an easy way to disambiguate I believe.